### PR TITLE
Adding support for Vector2f and HapticApply redo

### PR DESCRIPTION
**Pull request by peg-danielse**
sorry for the duplicate pull request somehow my branch got messed up by the formatting changes. If I remember correctly I fixed all the comments.

### CODE REVIEW

File: src/layer_shims.cpp [hunk 1]
```diff
@@ -432,12 +432,10 @@ void recordGetActionStateFloat(const XrActionStateGetInfo *getInfo, XrActionStat
     auto isActive = state->isActive;
     auto lastChanged = state->lastChangeTime;
 
-    // we don't need all values. especially if there are long periods that noting happens. check if we need to log
```
Comment (jdonkervliet) [inline src/layer_shims.cpp:452]: Any reason these comments are deleted? If they are outdated, could you replace, not delete? :)

**Review by jdonkervliet — CHANGES_REQUESTED** [review]
Other than the missing comments, looks good to me!

**Comment by jdonkervliet** [comment]
Can you resolve the conflict, @peg-danielse?

**Comment by peg-danielse** [comment]
There you go :)

**Pull request merged**